src/Microsoft.ServiceHub.Framework/ServiceHub/IDiscoveryService.cs

```diff
@@ -15,4 +15,11 @@ public interface IDiscoveryService
 	/// <param name="cancellationToken">A token to signal cancellation.</param>
 	/// <returns>The full path to the service's configuration file or null if the service was not found.</returns>
 	Task<string> DiscoverServiceAsync(string serviceName, CancellationToken cancellationToken);
+
+	/// <summary>
+	/// Find all configuration files for all services provided by this discovery service.
+	/// </summary>
+	/// <param name="cancellationToken">A token to signal cancellation.</param>
+	/// <returns>A collection of tuples containing the name of the service and the full path to the service's configuration file.</returns>
+	Task<IEnumerable<(string ServiceName, string FilePath)>> DiscoverAllServicesAsync(CancellationToken cancellationToken);
```

Comment:
Is this essentially an implementation detail of ServiceHub itself?

Reply:
> so that we can make sure that no new ones get added.

Do you mean... not without registering with your team? Why would you want to block all new services?

I guess I'm wondering whether you're concerned about this being a binary breaking change. Who all might implement this?

Reply:
We want to be able to block all new services that target .NET Framework so we can force everyone to migrate to .NET. Right now, only my team owns something that implements this but in theory anyone that hosts ServiceHub (Controller and Hosts) could

Reply:
And theoretically those other hosts (C# Dev Kit has one) wouldn't break until they took an update to servicehub, at which point they'd get a _compile_ error before they ever got to a runtime error, right?

Reply:
This would be a VS only thing, it wouldn't affect C# DevKit unless you would want to implement something similiar in the server process. This change just introduces the API that would enable the scenario